version: '3.8'

services:
  php-apache:
    container_name: app-faveo
    build: 
      context: .
      dockerfile: Dockerfile
    image: app-faveo
    networks:
      faveo-network:
        ipv4_address: 172.24.2.1
    ports:
      - "8282:80"
    volumes:
      - app-volume:/var/www/
    restart: always

  faveo-mariadb:
    image: mysql:8.0
    container_name: mysql-faveo
    volumes:
      - db-volume:/var/lib/mysql
      - ./src/init.sql:/docker-entrypoint-initdb.d/init.sql
    networks:
      faveo-network:
        ipv4_address: 172.24.2.2
    environment:
      MYSQL_DATABASE: faveo
      MYSQL_USER: faveo
      MYSQL_PASSWORD: Adm!202
      MYSQL_ROOT_PASSWORD: Adm!202
    restart: on-failure

  faveo-redis:
    image: redis
    container_name: redis-faveo
    restart: always
    networks:
      faveo-network:
        ipv4_address: 172.24.2.3

networks:
  faveo-network:
    name: faveo-network
    driver: bridge
    ipam:
      config:
        - subnet: 172.24.0.0/16

volumes:
  db-volume:
    name: db-volume
  app-volume:
    name: app-volume
    driver: local